osdep/timer.c

```diff
@@ -101,16 +101,26 @@ struct timespec mp_time_ns_to_realtime(int64_t time_ns)
     int64_t time_rel = time_ns;
 #endif
 
+    int64_t ns_1000_days = 1000 * 24 * 60 * 60 * INT64_C(1000000000);
+    assert(time_rel > -ns_1000_days && "Distant past conversion are not allowed!");
```

Comment:
you've built yourself a trap: `time_rel` is not relative on win32 so this check here or the clamp below might actually not do what's intended
(though likely no effect in practice)

is the shortcut still simpler than just doing it right? :)

Reply:
to the point: doesn't really hurt to allow distant past conversions, does it?
If so I'd just allow it.

Reply:
> `time_rel` is not relative on win32 so this check here or the clamp below might actually not do what's intended

Then there is not problem, absolute time will always be strictly positive.

Reply:
Sorry I meant just times in the past, clamping for future times is ok.

> How would you implement that? We have to clamp to some value, time_t type is not defined, do we clamp to INT64_MIN seconds instead? Does it work on macos? Why we want to support some imaginary timestamps from the past? What would be the use-case?

AFAICT it would suffice to readd this logic:
```
if (ts.tv_nsec < 0) {
        ts.tv_sec--;
        ts.tv_nsec += INT64_C(1000000000);
    }
```

and change nothing else.

But it also doesn't really matter.